src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/RootOperatorCostCache.java

```diff
@@ -0,0 +1,23 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RootOperatorCostCache {
+    protected final Map<PhysicalPlan, List<Integer>> map = new HashMap<>();
+
+    public void add ( final PhysicalPlan pp, final List<Integer> metrics ) {
```

Comment:
Please introduce an actual interface for the parameter 'metric', where this interface has a specific getter method for each of the five metrics, e.g., getNumberOfRequests().